pkg/api/v1/pod/util.go

```diff
@@ -48,30 +49,9 @@ func FindPort(pod *v1.Pod, svcPort *v1.ServicePort) (int, error) {
 	return 0, fmt.Errorf("no suitable port for manifest: %s", pod.UID)
 }
 
-// ContainerType signifies container type
-type ContainerType int
-
-const (
-	// Containers is for normal containers
-	Containers ContainerType = 1 << iota
-	// InitContainers is for init containers
-	InitContainers
-	// EphemeralContainers is for ephemeral containers
-	EphemeralContainers
-)
-
-// AllContainers specifies that all containers be visited
-const AllContainers ContainerType = InitContainers | Containers | EphemeralContainers
-
-// AllFeatureEnabledContainers returns a ContainerType mask which includes all container
-// types except for the ones guarded by feature gate.
-func AllFeatureEnabledContainers() ContainerType {
-	return AllContainers
-}
-
 // ContainerVisitor is called with each container spec, and returns true
 // if visiting should continue.
-type ContainerVisitor func(container *v1.Container, containerType ContainerType) (shouldContinue bool)
+type ContainerVisitor func(container *v1.Container, containerType resource.ContainerType) (shouldContinue bool)
```

Comment:
making the container iteration functions use constants from the resource helper package is really weird... I don't see the value of sharing types here. I think we should restore the ContainerType / Containers/InitContainers/EphemeralContainers / AllContainers / AllFeatureEnabledContainers bits to this package, and narrowly define the two constants we need in the resource helper package (Containers/InitContainers)


that would also drop out ~all the unrelated package import changes in this PR for things which aren't actually doing anything with resource calculation

Reply:
Updated :)